lib/common/api/deprecate.js

```diff
@@ -46,6 +46,26 @@ deprecate.log = (message) => {
   }
 }
 
+// Deprecate an event.
+deprecate.event = (emitter, oldName, newName, fn) => {
+  let warned = false
+  return emitter.on(newName, function (...args) {
+    // There are no listeners for this event
+    if (this.listenerCount(oldName) === 0) { return }
+    // noDeprecation set or if user has already been warned
+    if (warned || process.noDeprecation) { return }
```

Comment:
Can `process.noDeprecation` be changed in runtime?
If no, then maybe `deprecate.event()` shouldn't do anything if the flag is set?

Reply:
As far as I know, setting `process.noDeprecation` at runtime should work.